### DISCUSSION TITLE
Mount automatically /etc/fstab , Next Reboot ,how automatically mount

### DISCUSSION THREAD

**Discussion by Curiouserw** [juicedata/juicefs#2885] [Q&A]
I konw that how to mount by export META_PASSWORD environment to keep Metadata Engine password  securitly , at the same time i setup my SHELL environment don't record the command history。

I write mount command to /etc/fstab , then 

```bash
echo "mount commnad" >> /etc/fstab 
set +o history
export META_PASSWORD=***** && mount -a
```
But if ,the machine reboot 。the META_PASSWORD environment will does not exist，How to mount automatically （I don't want write export META_PASSWORD to /etc/profile ,it not security ）？

**Comment by xiaogaozi** [comment]
What operating system and what version are you using?

**Reply by Curiouserw** [reply]
> What operating system and what version are you using?

CentOS 7.7
juicefs version 1.0.0+2022-08-08.cf0c269

**Reply by Curiouserw** [reply]
我用中文描述问题吧。
   我比较关注客户端挂载时，元数据存储的安全问题。手动挂载和手动生效自动挂载时我都测试过，通过设置shell不记录命令执行过程，设置元数据密码的环境变量可以达到保护元数据存储用户密码进行挂载的目的。但是，如果在/etc/fstab设置自动挂载，那服务器自动重启的时候，是不是自动挂载就不生效了。

**Reply by Curiouserw** [reply]
同时，挂载和管理文件系统都用一个客户端进行管理，是不是客户端的功能权限太高了。

**Reply by xiaogaozi** [reply]
通过 `/etc/fstab` 的方式恐怕没法读取环境变量，可能得通过 systemd 来配置开机自动挂载。

**Reply by xyb** [reply]
感谢大家提供的线索，刚刚添加了 systemd.mount 的文档：https://github.com/juicedata/juicefs/commit/d3cfefa7f2c931262e35e837bd6a4641c98171cf ，有需要的话可以供参考。
